File: python/flask/src/app/modules/admin/module.py
# Commons
from app.common.classes.base_module import BaseModule

from .service import AdminService
from .controller import AdminController
from app.modules.managers.repository import ManagersRepository
from app.modules.clients.repository import ClientsRepository

class AdminModule(BaseModule):
    def __init__(self):
        _repositories = {
            'managers': ManagersRepository(),
            'clients': ClientsRepository(),
        }
        _service = AdminService(_repositories)
        self._controller = AdminController(_service)

        super().__init__(prefix="admin")

    def _routes(self):
        self._router.get(self._controller.find_all_clients, path='clientes')
        self._router.get(self._controller.find_all_managers, path="gerentes")

File: rust/axum/src/modules/admin/module.rs
use axum::{Router, routing::get};
use super::controller::AdminController;

pub fn admin_module() -> Router {
    let router = 
    Router::new()
    .route("/clientes", get(AdminController::find_all_clients))
    .route("/gerentes", get(AdminController::find_all_managers));

    Router::new().nest("/admin", router)
}
